pkg/binary/envoy/runtime.go

```diff
@@ -57,39 +58,33 @@ func (o RuntimeOptions) AndAll(opts RuntimeOptions) RuntimeOptions {
 }
 
 // NewRuntime creates a new Runtime with the local file storage set to the home directory
-func NewRuntime(options ...RuntimeOption) (binary.FetchRunner, error) {
-	local := common.HomeDir
+func NewRuntime(homeDir, debugDir string, options ...RuntimeOption) (binary.FetchRunner, error) {
+	// debugDir is intentionally lazy created as not all commands result in a running Envoy (ex fetch)
+	if debugDir == "" {
+		debugDir = filepath.Join(homeDir, "debug")
+	}
+
 	runtime := &Runtime{
-		RootDir:         local,
-		fetcher:         fetcher{local},
-		TmplDir:         filepath.Join(local, "templates"),
+		homeDir:         homeDir,
+		debugStore:      filepath.Join(debugDir, strconv.FormatInt(time.Now().UnixNano(), 10)),
```

Comment:
exact same pattern as before, just based over a variable. We can't easily undo the unixNano thing, and there's not a big need to change that as long as we can change the basedir